### PR TITLE
Adding a check whether the label is main or not

### PR DESCRIPTION
**Pull request by spiderjako**
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/model/project/project.ex [hunk 1]
```diff
@@ -376,6 +376,10 @@ defmodule Sanbase.Model.Project do
     end
   end
 
+  def has_main_contract_addresses(project) do
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project/project.ex:379]:
```suggestion
  def has_main_contract_addresses?(project) do
```

File: lib/sanbase/model/project/project.ex [hunk 2]
```diff
@@ -376,6 +376,10 @@ defmodule Sanbase.Model.Project do
...
+    project.contract_addresses |> Enum.findby(&(&1.label == "main"))
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project/project.ex:380]: You have to do it like this:
```
project
|> Repo.preload([:conract_addresses])
...
```
This is no-op when the contracts are preloaded, but makes it work when they're not, too.

Comment (IvanIvanoff) [reply lib/sanbase/model/project/project.ex:380]: Also what is `Enum.findby`? There's no` Enum.findby` and there's neither `Enum.find_by`.
Lastly, you want this function to return `true/false` (answers the question has?) and not `nil` or the contract address element.

**Comment by tspenov** [comment]
@spiderjako please include a test case for this change

File: lib/sanbase/model/project/project.ex [hunk 3]
```diff
@@ -376,6 +376,19 @@ defmodule Sanbase.Model.Project do
     end
   end
 
+  def has_main_contract_addresses?(project) do
+    project
+    |> IO.inspect()
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project/project.ex:381]: inspect left

File: lib/sanbase/model/project/project.ex [hunk 4]
```diff
@@ -376,6 +376,19 @@ defmodule Sanbase.Model.Project do
...
+    |> Repo.preload([:contract_addresses])
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project/project.ex:382]: Also think what this code is actually doing and why it's not correct:
```elixir
project
|> |> Repo.preload([:contract_addresses])
```

File: test/sanbase/external_services/project_info_test.exs [hunk 5]
```diff
@@ -137,7 +137,12 @@ defmodule Sanbase.ExternalServices.ProjectInfoTest do
     |> Repo.insert!()
 
     project =
-      %Project{slug: "santiment", name: "Santiment", ticker: "OLD_TICKR"}
+      %Project{
+        slug: "santiment",
+        name: "Santiment",
+        ticker: "OLD_TICKR",
+        contract_addresses: ["0x0000000", "0x0000001"]
+      }
```
Comment (IvanIvanoff) [inline test/sanbase/external_services/project_info_test.exs:145]: Use the factory to create a project with 2 contracts

Comment (IvanIvanoff) [reply test/sanbase/external_services/project_info_test.exs:145]: Hm, this test is for insertion of tags, not related to this PR.
You should add a test to the scraper tests where:
1. First insert a project and a `main` contract for it
2. Run the scraper
3. Assert the new contract is not labeled as main

File: lib/sanbase/model/project/project.ex [hunk 6]
```diff
@@ -376,6 +376,19 @@ defmodule Sanbase.Model.Project do
     end
   end
 
+  def has_main_contract_addresses?(project) do
+    preloaded_project =
+      project
+      |> Repo.preload([:contract_addresses])
+
+    preloaded_project.contract_addresses
+    |> Enum.find(&(&1.label == "main"))
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project/project.ex:385]:
```suggestion
    project
    |> Repo.preload([:contract_addresses])
    |> Map.get(:contract_addresses)
    |> Enum.find(&(&1.label == "main"))
```

File: test/sanbase/external_services/coinmarketcap/scraper_test.exs [hunk 7]
```diff
@@ -19,4 +22,25 @@ defmodule Sanbase.ExternalServices.Coinmarketcap.ScraperTest do
...
+    ordinary_contract = build(:contract_address)
+    third_contract = build(:contract_address)
+
+    project =
+      insert(:random_project,
+        contract_addresses: [main_contract, ordinary_contract]
+      )
+
+    project_info_map = %ProjectInfo{
+      slug: "santiment",
```
Comment (tspenov) [inline test/sanbase/external_services/coinmarketcap/scraper_test.exs:37]: here `santiment` slug is used whereas above a random project is build.

File: test/sanbase/external_services/coinmarketcap/scraper_test.exs [hunk 8]
```diff
@@ -19,4 +22,25 @@ defmodule Sanbase.ExternalServices.Coinmarketcap.ScraperTest do
...
+      main_contract_address: third_contract.address
+    }
+
+    File.read!(Path.join(__DIR__, "data/project_page.html"))
+    |> Scraper.parse_project_page(project_info_map)
```
Comment (tspenov) [inline test/sanbase/external_services/coinmarketcap/scraper_test.exs:42]: `parse_project_page` only updates `%ProjectInfo` struct - it doe not update the project and its contract addresses

File: test/sanbase/external_services/coinmarketcap/scraper_test.exs [hunk 9]
```diff
@@ -19,4 +22,25 @@ defmodule Sanbase.ExternalServices.Coinmarketcap.ScraperTest do
...
+
+    assert Enum.find(project.contract_addresses, &(&1.address == third_contract.label != "main"))
```
Comment (tspenov) [inline test/sanbase/external_services/coinmarketcap/scraper_test.exs:44]: I've never seen that  x==y==z used but seems not correct - `address` is compared with `label`and with string `main`. I guess the correct should be `&1.address == third_contract.address and &1.label != "main"`

**Review by tspenov — COMMENTED** [review]
One of the tests is not testing the behaviour correctly. I see that next text is almost the same so it also should be fixed.

**Review by IvanIvanoff — APPROVED** [review]

**Pull request merged**